### PR TITLE
allow same name column on export

### LINKED ISSUE #8: [Bug]: colums missing in CSV Export from GRID with multiple operators of the same type #11493
https://github.com/pimcore/pimcore/issues/11493

### PR DESCRIPTION
**Pull request by cameronfromtorq**
Resolves #8

Use a unique index for each column of the export array instead of indexing based on column name. 

If desired, I could also add an option to the export panel to set custom header names, personally I think this should be split into a separate feature request.

### CODE REVIEW

File: models/DataObject/Service.php [hunk 1]
```diff
@@ -1819,29 +1820,29 @@ public static function getCsvDataForObject(Concrete $object, $requestedLanguage,
...
                 }
 
-                $objectData[$field] = $fieldData;
+                $objectData[$index] = ['fieldName' => $field, 'data' => $fieldData];
             }
+            $index++;
         }
 
         if ($returnMappedFieldNames) {
             $tmp = [];
```
Comment (lukemacausland) [inline models/DataObject/Service.php]: This may not be used anymore?

**Review by lukemacausland — CHANGES_REQUESTED** [review]
Looks good, I think just that one comment.

File: models/DataObject/Service.php (hunk 1 above)
Comment (cameronfromtorq) [reply models/DataObject/Service.php:1842]: Yep! I just added it.

Comment (cameronfromtorq) [reply models/DataObject/Service.php]: The index is needed, but its better to use loop index so I changed it.

Comment (lukemacausland) [reply models/DataObject/Service.php]: @cameronfromtorq sorry I meant I don't think `$tmp` is being used anymore.

**Review by lukemacausland — DISMISSED** [review]

**Comment by lukemacausland** [comment]
@cameronfromtorq - can you update to 10.6

**Review by evanjamesjackson — APPROVED** [review]
Gr8

**Pull request closed without merging**